### PR TITLE
Update `split_suggester.py` to widen use cases

### PR DESCRIPTION
**Pull request by DRGN-DRC**
Added support for `.float` and `.double` (rather than just `.4byte`) for sdata2 parsing; thus, it can now work on files that `convert_data.py` has already operated on (this will verify its results). Now prints all literals used in each file at the top of the output file(s) in C (commented out by default). Also added preliminary support for parsing other file sections (rodata, bss, sbss, etc.).

### CODE REVIEW

File: tools/split_suggester.py [hunk 1]
```diff
@@ -29,30 +36,29 @@
...
     if sys.argv[2].lower().endswith(".map"):
         mapFilePath = sys.argv[2]
     else:
         print("Invalid input; the second argument is expected to be a .map file.")
-        exit(1)
+        sys.exit(1)
 elif len(sys.argv) > 3:
     print("Invalid input; got more than the expected one or two arguments.")
-    exit(1)
+    sys.exit(1)
```
Comment (ribbanya) [inline tools/split_suggester.py:53]: Please rip this out and use `argparse`.

File: tools/split_suggester.py [hunk 2]
```diff
@@ -513,17 +609,33 @@ def userConfirms():
 
 
 # Print results
-print("Total functions: " + str(len(functions)))
-print("Total floats:    " + str(len(floatTypes)))
-print("Text Section:    0x{:X} bytes".format(textBytes))
-print("Data Section:    0x{:X} bytes".format(dataBytes))
-print("sdata2 Section:  0x{:X} bytes".format(sdataBytes))
+print("  Total functions:            " + str(len(functions)))
```
Comment (ribbanya) [inline tools/split_suggester.py:612]: Please move the main execution stuff to an `if __name__ == "__main__"` block.

File: tools/split_suggester.py [hunk 3]
```diff
@@ -513,17 +609,33 @@ def userConfirms():
...
 if mapNames:
     print("\n\t(Function names found in the map file are shown to the right.)")
 fileCount = len(fileList)
 if debugging:
-    print("All floats data:")
+    print("\nAll floats data:")
     print(floatsData)
+    print("\nAll float types:")
+    print(floatTypes)
     print("")
```
Comment (ribbanya) [inline tools/split_suggester.py:639]: Please use a multiline format string literal for everything like this.

**Review by ribbanya — CHANGES_REQUESTED** [review]
Okay these are the changes I have for now. I'll look it over again after they're implemented.

File: tools/split_suggester.py (hunk 1 above)
Comment (DRGN-DRC) [reply tools/split_suggester.py]: Done.

File: tools/split_suggester.py (hunk 3 above)
Comment (DRGN-DRC) [reply tools/split_suggester.py]: Done.

File: tools/split_suggester.py (hunk 2 above)
Comment (DRGN-DRC) [reply tools/split_suggester.py]: Done.

**Review by ribbanya — COMMENTED** [review]
This is looking better. I can take a closer look soon.

**Review by ribbanya — CHANGES_REQUESTED** [review]
I replaced exit codes with exceptions. Still not the cleanest but it's okay for now.

Here are some of the issues that remain:

1. Paths assume Windows (backslash as separator). `os` can handle this but `pathlib` is better in general, so all path handling and `open` statements should be replaced with `pathlib` usage.
1. `global` should not be used. Pass arguments instead.
1. `str.format` should not be used. Use string interpolation instead.
1. `asmFile` should be a positional argument, not optional.
1. Python naming (snake case) needs to be used instead of camel case.
1. At this point, `split_suggester` should be in its own subdirectory and split into modules. 
1. Pyright errors (generated from ba2245ac029ef9c25e956549d7fdbd903fb9c351) need to be resolved:
    ```log
    ./tools/split_suggester.py
      ./tools/split_suggester.py:73:23 - error: Argument of type "int" cannot be assigne
    d to parameter "__x" of type "str | bytes | bytearray" in function "__new__"
        Type "int" cannot be assigned to type "str | bytes | bytearray"
          "int" is incompatible with "str"
          "int" is incompatible with "bytes"
          "int" is incompatible with "bytearray" (reportGeneralTypeIssues)
      ./tools/split_suggester.py:72:25 - error: Argument of type "int" cannot be assigne
    d to parameter "__x" of type "str | bytes | bytearray" in function "__new__"
        Type "int" cannot be assigned to type "str | bytes | bytearray"
          "int" is incompatible with "str"
          "int" is incompatible with "bytes"
          "int" is incompatible with "bytearray" (reportGeneralTypeIssues)
      ./tools/split_suggester.py:276:28 - error: "floatType" is possibly unbound (report
    UnboundVariable)
      ./tools/split_suggester.py:280:30 - error: "floatType" is possibly unbound (report
    UnboundVariable)
      ./tools/split_suggester.py:284:33 - error: Operator "+=" not supported for types "
    Unbound | bytes | bytearray" and "bytes"
        Operator "+" not supported for types "Unbound" and "bytes" (reportGeneralTypeIssues)
      ./tools/split_suggester.py:284:33 - error: "valueBytes" is possibly unbound (repor
    tUnboundVariable)
      ./tools/split_suggester.py:286:33 - error: Operator "+=" not supported for types "
    Unbound | bytes | bytearray" and "bytearray"
        Operator "+" not supported for types "Unbound" and "bytearray" (reportGeneralTypeIssues)
      ./tools/split_suggester.py:286:33 - error: "valueBytes" is possibly unbound (repor
    tUnboundVariable)
      ./tools/split_suggester.py:382:45 - error: "addLabel" is not a known member of "No
    ne" (reportOptionalMemberAccess)
      ./tools/split_suggester.py:385:45 - error: "addLabel" is not a known member of "No
    ne" (reportOptionalMemberAccess)
      ./tools/split_suggester.py:388:40 - error: "start" is not a known member of "None"
     (reportOptionalMemberAccess)
      ./tools/split_suggester.py:390:41 - error: "start" is not a known member of "None"
     (reportOptionalMemberAccess)
      ./tools/split_suggester.py:390:41 - error: Cannot assign member "start" for type "
    Function"
        Expression of type "str" cannot be assigned to member "start" of class "Function"
          "str" is incompatible with "int" (reportGeneralTypeIssues)
      ./tools/split_suggester.py:391:41 - error: "checkForMapName" is not a known member
     of "None" (reportOptionalMemberAccess)
      ./tools/split_suggester.py:395:37 - error: "lines" is not a known member of "None"
     (reportOptionalMemberAccess)
      ./tools/split_suggester.py:426:41 - error: "lines" is not a known member of "None"
     (reportOptionalMemberAccess)
      ./tools/split_suggester.py:725:33 - error: "previousFileIndex" is possibly unbound
     (reportUnboundVariable)
      ./tools/split_suggester.py:726:40 - error: "previousFileIndex" is possibly unbound
     (reportUnboundVariable)
      ./tools/split_suggester.py:730:47 - error: "previousFileIndex" is possibly unbound
     (reportUnboundVariable)
      ./tools/split_suggester.py:739:22 - error: "previousFileIndex" is possibly unbound
     (reportUnboundVariable)
      ./tools/split_suggester.py:739:52 - error: "previousFileIndex" is possibly unbound
     (reportUnboundVariable)
      ./tools/split_suggester.py:844:50 - error: "None" is not iterable (reportGeneralTy
    peIssues)
      ./tools/split_suggester.py:865:46 - error: "None" is not iterable (reportGeneralTy
    peIssues)
    23 errors, 0 warnings, 0 informations 
    Completed in 0.926sec
    ```

**Review by ribbanya — APPROVED** [review]
I'm going to merge this as-is but please keep in mind the principles in https://github.com/doldecomp/melee/pull/755#pullrequestreview-1335263150 when working with our Python code in the future.

**Comment by DRGN-DRC** [comment]
Thanks. It will still be useful to people as-is. (Assuming we mention it now and then so people know it exists.) Personally, I use it every time I start working on a file.

I still plan to come back to this later and clean it up some more.

**Pull request merged**